### PR TITLE
Support copy in Fluid channels

### PR DESCRIPTION
**Pull request by abhinavarora**
Fixes https://github.com/PaddlePaddle/Paddle/issues/9085

### CODE REVIEW

File: python/paddle/fluid/concurrency.py [hunk 1]
```diff
@@ -162,11 +164,26 @@ def channel_send(channel, value):
...
+        copied_X = helper.create_variable(
+            name=unique_name.generate(value.name + '_copy'),
+            type=value.type,
+            dtype=value.dtype,
+            shape=value.shape,
+            lod_level=value.lod_level,
+            capacity=value.capacity)
+
+        assign_op = channel_send_block.append_op(
+            type="assign_op", inputs={"X": value, }, outputs={"Out": copied_X})
```
Comment (cs2be) [inline python/paddle/fluid/concurrency.py]: although this is valid python syntax, I'm wondering if we can remove the extra "," after value

Comment (abhinavarora) [reply python/paddle/fluid/concurrency.py]: Done.

File: python/paddle/fluid/concurrency.py [hunk 2]
```diff
@@ -141,6 +141,8 @@ def channel_send(channel, value):
         channel (Variable|Channel): Channel variable created using
         `make_channel`.
         value (Variable): Value to send to channel
+        copy (bool): Copy data while channel send. If False, then data
+        is moved. The input cannot be used after move.
```
Comment (chengduoZH) [inline python/paddle/fluid/concurrency.py]: Please add the default value of `copy` at the end of line145.
Maybe replacing `copy` with `copy_mode` is better.

Comment (abhinavarora) [reply python/paddle/fluid/concurrency.py]: Sorry, I just realized that I missed your comment and merged the PR. I agree with you. Let me correct it and send you the review separately.

Comment (chengduoZH) [reply python/paddle/fluid/concurrency.py]: it doesn't matter, the current scenario for using channel is still less.

**Pull request merged**